envoy/stats/stats.h

```diff
@@ -101,6 +101,7 @@ class Metric : public RefcountInterface {
     static constexpr uint8_t Used = 0x01;
     static constexpr uint8_t LogicAccumulate = 0x02;
     static constexpr uint8_t NeverImport = 0x04;
+    static constexpr uint8_t Hidden = 0x08;
```

Comment:
It occurs to me that, as this applies only to gauges, we could encode the 4 possible states into 2 bits rather than using 4 bits.
```
00 = Uninitialized
01 = Accumulate
10 = NeverImport
11 = Hidden
```
or something like that. I'm not that concerned about saving the 2 flag bits but it's nicer not to have illegal states we should arguably check for.

I wanted to share this thought, but leaving the code as you have it is likely simpler than encoding as I suggested. But could you leave a note here that we are encoding 4 states in 4 bits and could easily collapse to 2 if we ever ran low?

Reply:
As per another comment, since a gauge can be both used and have different import modes. This would mean that it would have more than just 4 "states" so it can't be represented by 2 bits. And I worry that making it 3 bits to represent 8 states would start to make things a little complicated to read.

Reply:
The used-bit can stay as it is.

I was just talking about blocking out 2 bits for the 4 gauge-mode related states. It would be a little messy to code which is why I'm suggesting just a comment and an assert.